storage/storage-interface/src/state_view.rs

```diff
@@ -31,35 +29,36 @@ impl DbStateView {
 
 impl StateView for DbStateView {
     fn get_state_value(&self, state_key: &StateKey) -> Result<Option<Vec<u8>>> {
-        match state_key {
-            // Deal with the account state blob.
-            StateKey::AccessPath(path) => {
-                let blob_opt = self.get(&StateKey::AccountAddressKey(path.address))?;
-                if let Some(blob) = blob_opt {
-                    Ok(AccountState::try_from(&blob)?.get(&path.path).cloned())
-                } else {
-                    Ok(None)
-                }
-            }
-            _ => self.get(state_key),
-        }
+        self.get(state_key)
     }
 
     fn is_genesis(&self) -> bool {
         self.version.is_none()
     }
 }
 
-pub trait LatestDbStateView {
-    fn latest_state_view(&self) -> Result<DbStateView>;
+pub struct LatestDbStateView {
+    db: Arc<dyn DbReader>,
 }
 
-impl LatestDbStateView for Arc<dyn DbReader> {
-    fn latest_state_view(&self) -> Result<DbStateView> {
-        Ok(DbStateView {
-            db: self.clone(),
-            version: self.get_latest_version_option()?,
-        })
+impl LatestDbStateView {
+    pub fn new(db: Arc<dyn DbReader>) -> Self {
+        Self { db }
+    }
+}
+
+impl StateView for LatestDbStateView {
+    fn get_state_value(&self, state_key: &StateKey) -> Result<Option<Vec<u8>>> {
+        let value_opt = self.db.get_latest_state_value(state_key.clone())?;
+        if let Some(value) = value_opt {
+            Ok(value.maybe_bytes)
+        } else {
+            Ok(None)
+        }
+    }
```

Comment:
Please don't.. it's a crime to return inconsistent (meaning at different db versions) view of the db from something called a "StateView". I'd call it a bug. Why is this necessary?

Reply:
Good point - this was needed for some tests. Found another way to fix that.